trunk/src/utest/srs_utest.hpp

```diff
@@ -131,4 +131,147 @@ class MockProtectedBuffer
+//            });
+//        }
+//
+//        // The coroutine will be stopped and wait for it to terminate.
+//        // So maybe it won't execute all your code there.
+//
+// Enjoiy the sugar for coroutines.
+#define SRS_COROUTINE_GO_IMPL(context, id, code_block)                            \
+    class AnonymousCoroutineHandler##id : public ISrsCoroutineHandler             \
+    {                                                                             \
+    private:                                                                      \
+        SrsCoroutineChan *ctx_;                                                   \
+                                                                                  \
+    public:                                                                       \
+        AnonymousCoroutineHandler##id(SrsCoroutineChan *c)                        \
+        {                                                                         \
+            /* Copy the context so that we can pop it in different coroutines. */ \
+            ctx_ = c->copy();                                                     \
+        }                                                                         \
+        ~AnonymousCoroutineHandler##id()                                          \
+        {                                                                         \
+            srs_freep(ctx_);                                                      \
+        }                                                                         \
+                                                                                  \
+    public:                                                                       \
+        virtual srs_error_t cycle()                                               \
+        {                                                                         \
+            SrsCoroutineChan &ctx = *ctx_;                                        \
+            (void)ctx;                                                            \
+            code_block;                                                           \
+            return srs_success;                                                   \
+        }                                                                         \
+    };                                                                            \
+    AnonymousCoroutineHandler##id handler##id(context);                           \
+    SrsSTCoroutine st##id("anonymous", &handler##id);                             \
+    srs_error_t err_coroutine##id = st##id.start();                               \
+    srs_assert(err_coroutine##id == srs_success)
+
+// A helper to create a anonymous coroutine like goroutine in Go.
+// For example:
+//        SRS_COROUTINE_GO({
+//            srs_usleep(1 * SRS_UTIME_MILLISECONDS);
+//        });
+#define SRS_COROUTINE_GO(code_block) \
```

Comment:
is the Macro input var `id` is missing?
```suggestion
#define SRS_COROUTINE_GO(id, code_block) \
```

Reply:
This is defined for usecase for just one coroutine.